### PR TITLE
[#13] reuse open-api-workflows workflows

### PR DESCRIPTION
**Pull request by SonnyBA**
Fixes https://github.com/maykinmedia/open-api-framework/issues/13

Changes

Reuses open-api-workflows workflows to make CI jobs consistent across the different projects.

### CODE REVIEW

**Review by Coperh — APPROVED** [review]
Also looks fine.

Same comment about the `open-api-workflows` not being the main branch

**Review by stevenbal — APPROVED** [review]
Looks good, just a minor question regarding the workflow name

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -86,103 +87,37 @@ jobs:
...
-#          output: 'trivy-results-docker.sarif'
-#          ignore-unfixed: true
-#      - name: Upload results to GH Security tab
-#        uses: github/codeql-action/upload-sarif@v2
-#        with:
-#          sarif_file: 'trivy-results-docker.sarif'
-
-  publish:
+  open-api-ci:
+    uses: maykinmedia/open-api-workflows/.github/workflows/ci.yml@v1
```
[the remark this answers is not in the document]
Comment (SonnyBA) [reply .github/workflows/ci.yml:103]: Yeah, the `maykinmedia/setup-django-backend@v1.3` is an [composite action](https://docs.github.com/en/actions/sharing-automations/creating-actions/creating-a-composite-action). Where the changes in the PR use [reusable workflows](https://docs.github.com/en/actions/sharing-automations/reusing-workflows#creating-a-reusable-workflow).

**Comment by SonnyBA** [comment]
@joeribekker This repository still has the requirement of a successful previous "Docker image build" job run in order to merge any pull request. Since that job now [moved](https://github.com/maykinmedia/open-api-workflows/blob/main/.github/workflows/ci.yml#L77) to the open-api-workflows repository this needs to be changed (in order to merge this PR). The [documentation](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/managing-github-actions-settings-for-a-repository#allowing-select-actions-and-reusable-workflows-to-run) mentions (in step 4) how this could be configured.

**Review by annashamray — APPROVED** [review]

**Comment by annashamray** [comment]
@SonnyBA I've added `maykinmedia/*` into action permissions in the Settings, but it hasn't helped. The problem was in required statuses for the protected branch (`main`), Now it should work

**Pull request merged**